src/Application/FileProcessor.php

```diff
@@ -138,6 +151,32 @@ private function parseFileAndDecorateNodes(File $file): ?SystemError
         return null;
     }
 
+    /**
+     * @param array<string,true> $allFiles
+     */
+    private function cacheFileDependencies(File $file, array $allFiles): void
+    {
+        $fileDependencies = [];
+        $dependencyResolver = $this->dependencyResolver;
+        $this->simpleCallableNodeTraverser->traverseNodesWithCallable(
```

Comment:
We traverse all the nodes in this file, using the DependencyResolver to find the dependencies for each node and building a list of dependencies for the file